consensus/hotstuff/model/errors.go

```diff
@@ -238,13 +238,34 @@ func IsInvalidSignatureIncludedError(err error) bool {
 // This can happen because:
 //   - one or many signatures added via TrustedAdd are invalid to their respective public keys.
 //   - OR the signatures are valid but the public keys were forged to sum up to an identity public key.
-var InvalidAggregatedSignatureError = errors.New("aggregated signature is invalid (identity signature)")
+type InvalidAggregatedSignatureError struct {
+	error
+}
+
+func NewInvalidAggregatedSignatureError(err error) error {
+	return InvalidAggregatedSignatureError{err}
+}
+
+func NewInvalidAggregatedSignatureErrorf(msg string, args ...interface{}) error {
+	return InvalidAggregatedSignatureError{fmt.Errorf(msg, args...)}
+}
+
+func (e InvalidAggregatedSignatureError) Error() string { return e.error.Error() }
```

Comment:
We can remove this definition because it's implicitly defined when `error` is embedded in `InvalidAggregatedSignatureError`.

```suggestion
```

Reply:
Are you sure? I think in this case, you might loose some information about the wrapped errors. I just typed up this example:
https://go.dev/play/p/8-RBskzm4oX
but I don't want to dismiss the possibility that I have a bug in my code ... its already late. See this example. I also added some explanation, why the implementation behaves that way.

Reply:
You are right that `Unwrap()` is required, otherwise `errors.As` wouldn't work. What I meant in the comment is that `Error()` is not needed, and it is inherited from the embedded field. 
I've updated your code to show that by keeping `Unwrap()` and removing all `Error()` definitions (by keeping the types you used). https://go.dev/play/p/Kk0bvxkeuhn

Reply:
I was using unwrap to explain ... but you are right, the `Error()` method is subtly different, because the wrapped `error` actually contains the entire error string, and the wrapper does not add any additional text in this implementation. 💡